app/models/coin.rb

```diff
@@ -17,8 +16,11 @@ class Coin < ApplicationRecord
   accepts_nested_attributes_for :coin_excluded_countries, allow_destroy: true, reject_if: :all_blank
   accepts_nested_attributes_for :influencer_reviews, allow_destroy: true, reject_if: :all_blank
 
-  scope :find_by_symbol, -> (symbol) { where('lower(symbol) = ?', symbol.downcase).first }
   scope :top, -> (limit) { order(ranking: :asc).limit(limit) }
+  scope :upcoming, -> { where(ico_status: "upcoming") }
+  scope :active, -> { where(ico_status: "active") }
+  scope :ended, -> { where(ico_status: "ended") }
+  scope :listed, -> { where(ico_status: "listed") }
```

Comment:
This is the only bit I'd change - the scopes can be defined dynamically in the `ICO_STATUSES.each` block below.